#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.35 0 0 4.7
  position 0 3.78608 0
}
TexturedBackground {
  texture "mountains"
}
RectangleArena {
  translation 0 -5 0
  rotation -1 0 0 1.57079632679
  floorSize 10 10
  floorAppearance Parquetry{
    type "dark strip"
    colorOverride 0.393 0.597 0.146
  }
}
PointLight{
  attenuation 0 0 1
  intensity 1
  location 0 -4 -1
}
PointLight{
  attenuation 0 0 1
  intensity 1
  location -1 -4 1
}
PointLight{
  attenuation 0 0 1
  intensity 1
  location 1 -4 0.4
}
PointLight{
  attenuation 0 0 1
  intensity 1
  location -2 -4 -0.5
}
Ball{
  translation 0 -5 -1
  size 0.2 1 0.2
  radius 0.15
  mass 100
}
Ball{
  translation -1 -5 1
  size 0.2 1 0.2
  radius 0.15
  mass 100
}
Ball{
  translation 1 -5 0.4
  size 0.2 1 0.2
  radius 0.15
  mass 100
}
Ball{
  translation -1.8 -5 -0.5
  size 0.2 1 0.2
  radius 0.15
  mass 100
}
Robot {
  translation -2.5 -4.9 0
  rotation 0 1 0 1.57
  children [
    DistanceSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
    DistanceSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    LightSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_left"
    }
    LightSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_right"
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 647.1380516011527
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 1 0 0 6.253150268834896
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.305882 0.898039 0.25098
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.04
                  subdivision 24
                }
              }
            ]
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 658.5660414304194
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 1 0 0 5.114769483742438
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 647.1380548331853
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 1 0 0 6.253153500867447
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position 658.5660471062939
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 1 0 0 5.114775159616928
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "task9"
}

